File: dataset/Vanilla - Copy/OPENAI_gpt-5-2025-08-07/Task79.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <iomanip>
#include <stdexcept>

std::string encrypt(const std::string& message, const std::string& key) {
    if (key.empty()) throw std::invalid_argument("Key must not be empty");
    std::ostringstream ss;
    ss << std::uppercase << std::hex << std::setfill('0');
    for (size_t i = 0; i < message.size(); ++i) {
        unsigned char c = static_cast<unsigned char>(message[i]) ^ static_cast<unsigned char>(key[i % key.size()]);
        ss << std::setw(2) << static_cast<int>(c);
    }
    return ss.str();
}

int main() {
    std::cout << encrypt("HELLO WORLD", "KEY") << std::endl;
    std::cout << encrypt("Attack at dawn!", "secret") << std::endl;
    std::cout << encrypt("OpenAI", "GPT") << std::endl;
    std::cout << encrypt("The quick brown fox jumps over the lazy dog.", "cipher") << std::endl;
    std::cout << encrypt("Lorem ipsum dolor sit amet", "xyz") << std::endl;
    return 0;
}

File: dataset/Vanilla/OPENAI_gpt-5-2025-08-07/Task79.java
import java.nio.charset.StandardCharsets;

public class Task79 {
    public static String encrypt(String message, String key) {
        byte[] mb = message.getBytes(StandardCharsets.UTF_8);
        byte[] kb = key.getBytes(StandardCharsets.UTF_8);
        if (kb.length == 0) throw new IllegalArgumentException("Key must not be empty");
        byte[] out = new byte[mb.length];
        for (int i = 0; i < mb.length; i++) {
            out[i] = (byte) (mb[i] ^ kb[i % kb.length]);
        }
        StringBuilder sb = new StringBuilder(out.length * 2);
        for (byte b : out) {
            sb.append(String.format("%02X", b));
        }
        return sb.toString();
    }

    public static void main(String[] args) {
        System.out.println(encrypt("HELLO WORLD", "KEY"));
        System.out.println(encrypt("Attack at dawn!", "secret"));
        System.out.println(encrypt("OpenAI", "GPT"));
        System.out.println(encrypt("The quick brown fox jumps over the lazy dog.", "cipher"));
        System.out.println(encrypt("Lorem ipsum dolor sit amet", "xyz"));
    }
}